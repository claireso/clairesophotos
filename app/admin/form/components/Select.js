import React from 'react'

export default (props) => {
  return (
    <div className="form__item form__item--inline">
      <label 
        className="form__label"
        htmlFor={ props.name }
      >
        { props.label }
      </label>
      <select 
        id={ props.name } 
        name={ props.name } 
        defaultValue={ props.value }
        required 
      >
        { props.options.map((option, index) => {
          return (
            <option 
              key={ index }
              value={ option.value }
            >
              { option.label }
            </option>
          )
        }) }
      </select>
    </div>
  )
}