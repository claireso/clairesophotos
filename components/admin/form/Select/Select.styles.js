import styled from 'styled-components'

export const SelectWrapper = styled.div`
  position: relative;
  svg {
    position: absolute;
    top: 50%;
    right: 0.9rem;
    bottom: 0;
    transform: translate(0, -50%) rotate(90deg);
    pointer-events: none;
  }
`

export const StyledSelect = styled.select`
  appearance: none;
  background: var(--white);
  border: 1px solid var(--gray-1);
  border-radius: 0.5rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  cursor: pointer;
  font-family: 'Roboto', Arial, sans-serif;
  font-size: 1.4rem;
  outline: none;
  padding: 0.8rem 2.6rem 0.8rem 0.8rem;
  &:focus {
    border-color: var(--gray-2);
  }
`
