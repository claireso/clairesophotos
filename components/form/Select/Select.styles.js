import styled from 'styled-components'

export const SelectWrapper = styled.div`
  position: relative;
  svg {
    position: absolute;
    top: 50%;
    right: 0.9rem;
    bottom: 0;
    transform: translate(0, -50%) rotate(90deg);
    pointer-events: none;
  }
`

export const StyledSelect = styled.select`
  appearance: none;
  background: var(--form-input-bg);
  border: 1px solid var(--border-color);
  border-radius: 0.5rem;
  box-shadow: 1px 1px 3px var(--box-shadow);
  cursor: pointer;
  font-family: 'Roboto', Arial, sans-serif;
  font-size: 1.4rem;
  outline: none;
  padding: 0.8rem 2.6rem 0.8rem 0.8rem;
  &:focus {
    border-color: var(--border-color-focus);
  }
`
