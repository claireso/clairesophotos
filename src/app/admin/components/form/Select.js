import React from 'react'
import PropTypes from 'prop-types'

const Select = (props = {}) => {
  return (
    <div className="form__item form__item--inline">
      <label className="form__label" htmlFor={props.name}>
        {props.label}
      </label>
      <select
        id={props.name}
        name={props.name}
        defaultValue={props.value}
        required
      >
        {props.options.map((option, index) => {
          return (
            <option key={index} value={option.value}>
              {option.label}
            </option>
          )
        })}
      </select>
    </div>
  )
}

Select.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ).isRequired,
  value: PropTypes.string
}

export default Select
