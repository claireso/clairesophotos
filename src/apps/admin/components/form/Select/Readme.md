### Example

```js
<Select
  label="Position"
  name="position"
  options={[
    {
      label: 'Center',
      value: 'center'
    },
    {
      label: 'Left',
      value: 'left'
    },
    {
      label: 'Right',
      value: 'right'
    }
  ]}
/>
```