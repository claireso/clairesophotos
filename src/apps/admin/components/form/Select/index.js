import React from 'react'
import PropTypes from 'prop-types'

import { GroupInline } from '../components/Group'
import Label from '../components/Label'

const Select = (props = {}) => {
  return (
    <GroupInline>
      <Label htmlFor={props.name}>{props.label}</Label>
      <select
        id={props.name}
        name={props.name}
        defaultValue={props.value}
        required
      >
        {props.options.map((option, index) => {
          return (
            <option key={index} value={option.value}>
              {option.label}
            </option>
          )
        })}
      </select>
    </GroupInline>
  )
}

Select.propTypes = {
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired
    })
  ).isRequired,
  value: PropTypes.string
}

export default Select
