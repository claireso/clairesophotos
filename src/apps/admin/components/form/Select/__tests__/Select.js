import React from 'react'
import { render } from 'react-testing-library'

import Select from '../index'

describe('<Select />', () => {
  test('should render select', () => {
    const props = {
      label: 'My select',
      name: 'myselect',
      options: [
        {
          label: 'option 1',
          value: 'option1'
        },
        {
          label: 'option 2',
          value: 'option2'
        }
      ],
      value: 'option2'
    }

    const { container } = render(<Select {...props} />)

    expect(container).toMatchSnapshot()
  })
})
