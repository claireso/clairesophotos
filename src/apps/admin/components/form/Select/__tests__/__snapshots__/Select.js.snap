// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render select 1`] = `
.c0 {
  margin: 0 0 var(--gutter);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > label {
  margin: 0 0.5rem 0 0;
}

.c1 {
  display: block;
  font-size: 1.6rem;
  margin: 0 0 1rem;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
      for="myselect"
    >
      My select
    </label>
    <select
      id="myselect"
      name="myselect"
      required=""
    >
      <option
        value="option1"
      >
        option 1
      </option>
      <option
        selected=""
        value="option2"
      >
        option 2
      </option>
    </select>
  </div>
</div>
`;
